tools/git-hooks/Readme.md

```diff
@@ -0,0 +1,33 @@
+# Git Hooks for iceoryx
+
+When developing on iceoryx, git-hooks are doing some sanity checks and can execute linters/formatter when you commit code to iceoryx.
+In the current state the git hooks do the following job:
+
+- checking if commit messages follows the format `[iox-#<Github issue number>]` and auto-format them if necessary
+- check if commit message subject and body follow six of the [seven rules of a great commit message](https://chris.beams.io/posts/git-commit/)
+- remove trailing whitespace from all modified files
+- run `clang-format` on all modified files
+
+For running `clang-format` it checks if the version 10 is used, otherwise it will print a warning,
```

Comment:
The big question is how compatible clang-format is within this versions or if a different output is generated. On ubuntu 20.04 there is also clang-format-11 available, just in case we want a specific version.

Reply:
I would say that version 10 is sufficient for us. It is available in Ubuntu 18, 20 and 21 and can be the "base" version.
After scanning the complete repo with clang-format-10 i just had 3 files which had some styling finding and pushed them here. Additionally i enabled some functions in the clang-format file.
After scanning the whole repo additionally with clang format 11, 12, 13 i can only see one file which differs from version 10 (function_ref.inl). With that, I agree that we can use versions up from 10.

Reply:
Ah thats great :D